disnake/state.py

```diff
@@ -214,67 +223,65 @@ def __init__(
         self.hooks: Dict[str, Callable] = hooks
         self.shard_count: Optional[int] = None
         self._ready_task: Optional[asyncio.Task] = None
-        self.application_id: Optional[int] = utils._get_as_snowflake(options, "application_id")
-        self.heartbeat_timeout: float = options.get("heartbeat_timeout", 60.0)
-        self.guild_ready_timeout: float = options.get("guild_ready_timeout", 2.0)
+        self.application_id: Optional[int] = None if application_id is None else int(application_id)
+        self.heartbeat_timeout: float = heartbeat_timeout
+        self.guild_ready_timeout: float = guild_ready_timeout
         if self.guild_ready_timeout < 0:
             raise ValueError("guild_ready_timeout cannot be negative")
 
-        allowed_mentions = options.get("allowed_mentions")
-
         if allowed_mentions is not None and not isinstance(allowed_mentions, AllowedMentions):
             raise TypeError("allowed_mentions parameter must be AllowedMentions")
 
         self.allowed_mentions: Optional[AllowedMentions] = allowed_mentions
         self._chunk_requests: Dict[Union[int, str], ChunkRequest] = {}
 
-        activity = options.get("activity", None)
         if activity:
             if not isinstance(activity, BaseActivity):
                 raise TypeError("activity parameter must derive from BaseActivity.")
 
-            activity = activity.to_dict()
+            self._activity: Optional[ActivityPayload] = activity.to_dict()
+        else:
+            self._activity: Optional[ActivityPayload] = None
 
-        status = options.get("status", None)
+        self._status: Optional[str] = None
         if status:
-            if status is Status.offline:
-                status = "invisible"
-            else:
-                status = str(status)
+            self._status = "invisible" if status is Status.offline else str(status)
 
-        intents = options.get("intents", None)
         if intents is not None:
             if not isinstance(intents, Intents):
                 raise TypeError(f"intents parameter must be Intent not {type(intents)!r}")
```

Comment:
```suggestion
                raise TypeError(f"intents parameter must be Intents, not {type(intents)!r}")
```
typo unrelated to this PR, just noticed it while reading through the changes once more

Reply:
Done in 5f5664afe52c96f3fe228260735f46701685b0b2.